lib/gush/worker.rb

```diff
@@ -51,7 +51,17 @@ def setup_job(workflow_id, job_id)
     def incoming_payloads
       payloads = {}
       job.incoming.each do |job_name|
-        payloads[job_name] = client.load_job(workflow.id, job_name).output_payload
+       job = client.load_job(workflow.id, job_name)
+       payloads[job.class.to_s] ||= []
+
+       if workflow.nameize_payloads
+         payload = {:id => job.name, :payload => job.output_payload}
```

Comment:
I think having two formats available for users will be confusing, so I'm fine with having this simplified and always returning hashes with `:id` and :`payload` keys. Even less code to maintain and definitely much less confusion along the way to the developers :)